libcxx/include/__locale

```diff
@@ -384,7 +384,7 @@ public:
     static const mask punct  = _ISpunct;
     static const mask xdigit = _ISxdigit;
     static const mask blank  = _ISblank;
-#if defined(__mips__)
+#if defined(__mips__) || (BYTE_ORDER == BIG_ENDIAN)
```

Comment:
I can't seem to find where the character mask you mention switches on `BYTE_ORDER` -- can you point out to me where this happens?

Reply:
When I run libcxx test, I test the following cases on arm64-be and arm64-le:
>std::regex_traits<char> t;
std::string s("w");    
assert(!t.isctype('-', t.lookup_classname(s.begin(), s.end())));
assert(!t.isctype('@', t.lookup_classname(s.begin(), s.end()))); 

These cases come from `libcxx/test/std/re/re.traits/isctype.pass.cpp`. They all fail on arm64be but succeed in arm64le.

Reply:
Obtaining the character mask in the function 'isctype'. It gets the character mask by reading compiler character mask table. The compiler character mask table depends on whether the platform is arm64be or arm64le.

Reply:
Yes, that's it. Getting the character mask on arm64be and arm64le will run this function`_LIBCPP_GET_C_LOCALE->__ctype_b`. The character mask obtained from this is dependent on the endianness of current platform. For example,  "-', it's character mask is 0000010011000000 on arm64e while on arm64le is 1100000000000100